src/displayapp/DisplayApp.cpp

```diff
@@ -146,17 +137,25 @@ void DisplayApp::Start(System::BootErrors error) {
 void DisplayApp::Process(void* instance) {
   auto* app = static_cast<DisplayApp*>(instance);
   NRF_LOG_INFO("displayapp task started!");
-  app->InitHw();
+  app->Init();
+
+  if (app->bootError == System::BootErrors::TouchController) {
+    app->LoadNewScreen(Apps::Error, DisplayApp::FullRefreshDirections::None);
+  } else {
+    app->LoadNewScreen(Apps::Clock, DisplayApp::FullRefreshDirections::None);
+  }
 
   while (true) {
     app->Refresh();
   }
 }
 
-void DisplayApp::InitHw() {
+void DisplayApp::Init() {
+  lcd.Init();
```

Comment:
any reasoning on why to change the order of the init calls? Or doesn't the order matter?

Reply:
I thought it made sense to initialise all the hardware first (LCD etc) and then initialise bits relying on it like LVGL afterwards. Before LVGL was loading a screen before lcd.Init() which didn't make any sense to me